### ISSUE TITLE
spring.application.name is set to "bootstrap" on 2.0

### ISSUE THREAD

**Issue by mp911de** [spring-cloud/spring-cloud-commons#214]
https://github.com/spring-cloud/spring-cloud-commons/commit/80555b3354cec70b5bb8ac29f5ccb47198838532#diff-ffd4ed2e59b54e6acb4b4f3a088f9e31 introduced a change to `BootstrapApplicationListener` that `spring.application.name` is set always to the value of `spring.config.name` which is `bootstrap` by default.

It's no longer possible to set `spring.application.name` via the bootstrap config and it's not possible to set the property using test properties (via `@SpringBootTest(properties = …)`).

**Comment by spencergibb** [comment]
@dsyer any reason you did that specifically?

**Comment by spencergibb** [comment]
@mp911de looking at the diff, it was always set (just in a different place) https://github.com/spring-cloud/spring-cloud-commons/commit/80555b3354cec70b5bb8ac29f5ccb47198838532#diff-ffd4ed2e59b54e6acb4b4f3a088f9e31L176

But prior, it was removed https://github.com/spring-cloud/spring-cloud-commons/commit/80555b3354cec70b5bb8ac29f5ccb47198838532#diff-ffd4ed2e59b54e6acb4b4f3a088f9e31L201

**Comment by spencergibb** [comment]
So I guess the question now is, @dsyer why not remove spring.application.name anymore?

**Comment by spencergibb** [comment]
Putting the removal of `spring.application.name` back doesn't break commons.

**Comment by spencergibb** [comment]
@mp911de can you be more specific. Spring Cloud {Netflix|Consul} sets `spring.application.name` in various places and it works just fine.

**Comment by mp911de** [comment]
Resolving `spring.application.name` via `Environment` from boostrap context components [resolves in 2.0](https://github.com/spring-cloud/spring-cloud-vault/blob/2.0.x/spring-cloud-vault-config/src/main/java/org/springframework/cloud/vault/config/VaultProperties.java#L105) always to `bootstrap` (It still works for 1.x).

My `PropertySources` at resolution time:

```
0 = {org.springframework.boot.context.properties.source.ConfigurationPropertySourcesPropertySource@3329} "ConfigurationPropertySourcesPropertySource [name='configurationProperties']"
1 = {org.springframework.core.env.MapPropertySource@3330} "MapPropertySource [name='bootstrap']"
2 = {org.springframework.core.env.MapPropertySource@3331} "MapPropertySource [name='systemProperties']"
3 = {org.springframework.boot.env.OriginTrackedSystemPropertySource@3332} "OriginTrackedSystemPropertySource [name='systemEnvironment']"
4 = {org.springframework.boot.env.RandomValuePropertySource@3333} "RandomValuePropertySource [name='random']"
5 = {org.springframework.core.env.MapPropertySource@3334} "MapPropertySource [name='springCloudClientHostInfo']"
6 = {org.springframework.boot.env.OriginTrackedMapPropertySource@3335} "OriginTrackedMapPropertySource [name='applicationConfig: [classpath:/bootstrap.properties]']"
```

The `PropertySource` `bootstrap` at `1` contains `spring.config.name=bootstrap` and `spring.application.name=bootstrap`. The last  `PropertySource` `OriginTrackedMapPropertySource` at `6` contains the values from my `bootstrap.properties`.

Because of the order (it's a bit obfuscated by `ConfigurationPropertySourcesPropertySource`), `spring.application.name` resolves to `bootstrap` and not the value set in  `bootstrap.properties`

**Comment by spencergibb** [comment]
So you're seeing this in `org.springframework.cloud.bootstrap.BootstrapConfiguration`?

**Comment by mp911de** [comment]
Yes, it's during `BootstrapConfiguration`.

**Comment by spencergibb** [comment]
@mp911de can you try with snapshots and see if 	b5f5912 fixes your issue?

**Comment by mp911de** [comment]
Gave it a try but the behavior persists. I took a look on the code on `master` to see the difference. 

`spring.application.name` set by `BootstrapApplicationListener` is available during `SpringApplicationBuilder.run()` in a `PropertySource` which is the last property source. Other property sources (like the test properties or `boostrap.yml`) have a higher priority and that's why it works for `master`.

**Comment by spencergibb** [comment]
Closed via ed91ad8
